lib/core/test/element_special_values_test.cpp

```diff
@@ -312,3 +313,53 @@ TYPED_TEST(ElementNegativeInfToNumTest, value_and_variance_out) {
   targetted_replacement_out_arg_test(element::negative_inf_to_num_out_arg, out,
                                      replaceable, nonreplaceable, replacement);
 }
+
+TEST(ElementSpecialValueVector3dTest, isnan_true_if_one_element_is_nan) {
+  Eigen::Vector3d v(1.0, 2.0, std::numeric_limits<double>::quiet_NaN());
+  EXPECT_TRUE(element::isnan(v));
+}
+
+TEST(ElementSpecialValueVector3dTest, isnan_false_if_no_element_is_nan) {
+  Eigen::Vector3d v(1.0, 2.0, 3.0);
+  EXPECT_FALSE(element::isnan(v));
+}
+
+TEST(ElementSpecialValueVector3dTest, isfinite_false_if_one_element_is_nan) {
+  Eigen::Vector3d v(1.0, 2.0, std::numeric_limits<double>::quiet_NaN());
+  EXPECT_FALSE(element::isfinite(v));
+}
+
+TEST(ElementSpecialValueVector3dTest, isfinite_true_if_no_element_is_nan) {
+  Eigen::Vector3d v(1.0, 2.0, 3.0);
+  EXPECT_TRUE(element::isfinite(v));
+}
+
+TEST(ElementSpecialValueVector3dTest, isinf_false_if_one_element_is_nan) {
+  Eigen::Vector3d v(1.0, 2.0, std::numeric_limits<double>::quiet_NaN());
+  EXPECT_FALSE(element::isinf(v));
+}
+
+TEST(ElementSpecialValueVector3dTest, isnan_false_if_one_element_is_inf) {
+  Eigen::Vector3d v(1.0, 2.0, std::numeric_limits<double>::infinity());
+  EXPECT_FALSE(element::isnan(v));
+}
+
+TEST(ElementSpecialValueVector3dTest, isfinite_false_if_one_element_is_inf) {
+  Eigen::Vector3d v(1.0, 2.0, std::numeric_limits<double>::infinity());
+  EXPECT_FALSE(element::isfinite(v));
+}
+
+TEST(ElementSpecialValueVector3dTest, isfinite_true_if_no_element_is_inf) {
+  Eigen::Vector3d v(1.0, 2.0, 3.0);
+  EXPECT_TRUE(element::isfinite(v));
+}
+
+TEST(ElementSpecialValueVector3dTest, isinf_true_if_one_element_is_inf) {
+  Eigen::Vector3d v(1.0, 2.0, std::numeric_limits<double>::infinity());
+  EXPECT_TRUE(element::isinf(v));
+}
+
+TEST(ElementSpecialValueVector3dTest, isinf_false_if_no_element_is_inf) {
+  Eigen::Vector3d v(1.0, 2.0, 3.0);
+  EXPECT_FALSE(element::isinf(v));
+}
```

Comment:
Add `isinf_false_if_one_element_is_nan`?

Reply:
It is in line 337?

Reply:
Oh, right. In my head I was grouping them differently than their order in the file.